AdminCore.Services/EventService.cs

```diff
@@ -146,17 +147,51 @@ public EventDto CreateEvent(EventDateDto dates, EventTypes eventTypes, Employee
       return _mapper.Map<EventDto>(insertedEvent);
     }
 
-    public void UpdateEvent(EventDateDto eventDateDto, string message, int employeeId)
+    public void UpdateEvent(EventDateDto dates, string message, int employeeId)
+    { 
+      var eventToUpdate = GetEventById(dates.EventId);
+      var currentDates = eventToUpdate.EventDates;
+
+      if (EventIsNotUpdatable(eventToUpdate, message))
+      {
+        throw new Exception("Cannot update event, criteria not met");
+      }
+
+      EvaluateEventDates(dates, currentDates, eventToUpdate);
```

Comment:
Is there a check for if the if EventType or Half-Day state changes?

Reply:
Good shout with checking `Half-Day` state, forgot about it, EventType state is not supported. In fact, it is only visible in front end, the `UpdateEventViewModel` does not handle `EventType`. I have raised [LM-45](https://unosquarebelfast.atlassian.net/browse/LM-45) to add support for this